backend/service/k8s/event.go

```diff
@@ -0,0 +1,61 @@
+package k8s
+
+import (
+	"context"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	k8sapiv1 "github.com/lyft/clutch/backend/api/k8s/v1"
+)
+
+func (s *svc) ListEvents(ctx context.Context, clientset, cluster, namespace, object string, kind k8sapiv1.ObjectKind) ([]*k8sapiv1.Event, error) {
+	cs, err := s.manager.GetK8sClientset(ctx, clientset, cluster, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	objKind := strcase.ToCamel(strings.ToLower(kind.String()))
+	// returns the appropriate field selector based on the object involved
+	fieldSelector := cs.CoreV1().Events(cs.Namespace()).GetFieldSelector(&object, &namespace, &objKind, nil)
+
+	eventList, err := cs.CoreV1().Events(cs.Namespace()).List(ctx, metav1.ListOptions{FieldSelector: fieldSelector.String()})
+	if err != nil {
+		return nil, err
+	}
+
+	var events []*k8sapiv1.Event
+	for _, e := range eventList.Items {
+		event := e
+		events = append(events, ProtoForEvent(cs.Cluster(), &event))
+	}
+
+	return events, nil
+}
+
+func ProtoForEvent(cluster string, k8sEvent *corev1.Event) *k8sapiv1.Event {
+	clusterName := k8sEvent.ClusterName
+	if clusterName == "" {
+		clusterName = cluster
+	}
+	return &k8sapiv1.Event{
+		Cluster:        clusterName,
+		Namespace:      k8sEvent.Namespace,
+		Name:           k8sEvent.Name,
+		Reason:         k8sEvent.Reason,
+		Description:    k8sEvent.Message,
+		InvolvedObject: k8sEvent.InvolvedObject.Name,
+		Kind:           protoForObjectKind(k8sEvent.InvolvedObject.Kind),
```

Comment:
do you think we'll need the other fields in the InvolvedObject struct in the future? https://pkg.go.dev/k8s.io/api/core/v1#ObjectReference. if so, maybe we should define a proto for it

if not, since we're not grabbing all the fields, I think we should rename `InvolvedObject` to include Name for clarity

Reply:
For now there is no other use case for it anywhere else in the clutch code,  it is just a way for us to associate an event with a K8s object.

I can update the field name to be `InvolvedObjectName`

Reply:
Updated to `InvolvedObjectName`